config/webpack/development.js

```diff
@@ -1,23 +1,21 @@
 process.env.NODE_ENV = process.env.NODE_ENV || 'development'
 
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin')
+const path = require('path')
 const { devServer } = require('@rails/webpacker')
-const webpackConfig = require('./webpackConfig')
-
-module.exports = webpackConfig()
 
-const developmentOnly = () => {
-  const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin')
-  const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin')
-  const path = require('path')
+const webpackConfig = require('./webpackConfig')
 
-  const environment = require('./environment')
+const developmentEnvOnly = (clientWebpackConfig, serverWebpackConfig) => {
 
   const isWebpackDevServer = process.env.WEBPACK_DEV_SERVER
 
   //plugins
   if (isWebpackDevServer) {
-    environment.plugins.append(
-      'ReactRefreshWebpackPlugin',
+    // Note, when this is run, we're building the server and client bundles in separate processes.
+    // Thus, this plugin is not applied.
+    const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin')
+    clientWebpackConfig.plugins.push(
```

Comment:
This may require updates.

See discussion here: https://github.com/pmmmwh/react-refresh-webpack-plugin/issues/252#issuecomment-776121134

At the min, you'll need https://www.npmjs.com/package/@pmmmwh/react-refresh-webpack-plugin/v/next

Check other open and closed issues as well.